src/main/java/io/ticktok/Ticktok.java

```diff
@@ -4,21 +4,33 @@
 import io.ticktok.register.Clock;
 import io.ticktok.rest.RestTicktokClient;
 
+import static io.ticktok.validator.TicktokValidator.validate;
+
 public class Ticktok {
 
     private TicktokOptions options;
+    private String name;
+    private String schedule;
 
     public Ticktok(TicktokOptions options){
         this.options = options;
     }
 
-    public void newClock(String schedule, Runnable runnable) {
-        try {
-            Clock clock = new RestTicktokClient(this.options).register(schedule);
-            TickListener.listen(clock.getClockChannel(), runnable);
-        } catch (Exception e) {
-            throw new TicktokException("operation failed: " + e);
-        }
+    public Ticktok newClock(String name) {
+        validate(name);
```

Comment:
validations should come in the build function and not very helpful here if no one called this function

Reply:
you should validate when you do the action, here is on invoke 
as the validation wont work if I never called the function at all